files/scripts/swss.sh

```diff
@@ -10,67 +57,55 @@ start() {
     until [[ $(/usr/bin/docker exec database redis-cli -n 4 GET "CONFIG_DB_INITIALIZED") ]];
         do sleep 1;
     done
+}
+
+start() {
+    debug "Starting ${SERVICE} service..."
+
+    lock_service_state_change
 
-    SYSTEM_WARM_START=`/usr/bin/docker exec database redis-cli -n 4 HGET "WARM_RESTART|system" enable`
-    SWSS_WARM_START=`/usr/bin/docker exec database redis-cli -n 4 HGET "WARM_RESTART|swss" enable`
-    # if warm start enabled, just do swss docker start.
-    # Don't flush DB or try to start other modules.
-    if [[ "$SYSTEM_WARM_START" == "true" ]] || [[ "$SWSS_WARM_START" == "true" ]]; then
-      RESTART_COUNT=`redis-cli -n 6 hget "WARM_RESTART_TABLE|orchagent" restart_count`
-      # We have to make sure db data has not been flushed.
-      if [[ -n "$RESTART_COUNT" ]]; then
-        /usr/bin/swss.sh start
-        /usr/bin/swss.sh attach
-        return 0
-      fi
+    wait_for_database_service
+    check_warm_boot
+    validate_restart_count
+
+    debug "Warm boot flag: ${SERVICE} ${WARM_BOOT}."
+
+    # Don't flush DB during warm boot
+    if [[ x"$WARM_BOOT" != x"true" ]]; then
+        /usr/bin/docker exec database redis-cli -n 0 FLUSHDB
+        /usr/bin/docker exec database redis-cli -n 2 FLUSHDB
+        /usr/bin/docker exec database redis-cli -n 5 FLUSHDB
+        /usr/bin/docker exec database redis-cli -n 6 FLUSHDB
     fi
 
-    # Flush DB
-    /usr/bin/docker exec database redis-cli -n 0 FLUSHDB
-    /usr/bin/docker exec database redis-cli -n 1 FLUSHDB
-    /usr/bin/docker exec database redis-cli -n 2 FLUSHDB
-    /usr/bin/docker exec database redis-cli -n 5 FLUSHDB
-    /usr/bin/docker exec database redis-cli -n 6 FLUSHDB
-
-    # platform specific tasks
-    if [ x$sonic_asic_platform == x'mellanox' ]; then
-        FAST_BOOT=1
-        /usr/bin/mst start
-        /usr/bin/mlnx-fw-upgrade.sh
-        /etc/init.d/sxdkernel start
-        /sbin/modprobe i2c-dev
-        /etc/mlnx/mlnx-hw-management start
-    elif [ x$sonic_asic_platform == x'cavium' ]; then
-        /etc/init.d/xpnet.sh start
+    # start service docker
+    /usr/bin/${SERVICE}.sh start
+    debug "Started ${SERVICE} service..."
+
+    if [[ x"$WARM_BOOT" != x"true" ]] && [[ -z "$peerlock" ]]; then
```

Comment:
>$peerlock [](start = 48, length = 9)

in line 65, you obtained lock for both swss and syncd, why check peerlock here? why not just restart syncd service here?